packages/chain-adapters/src/evm/utils.ts

```diff
@@ -12,3 +14,56 @@ export const getErc20Data = async (
   const { data: callData } = await erc20Contract.populateTransaction.transfer(to, value)
   return callData || ''
 }
+
+type Eip1559GasPrice = {
+  fast?: string
+  average?: string
+  slow?: string
+}
+
+// calculate eip1559 gas price (baseFee + maxPriorityFeePerGas)
+// calculate baseFee ((maxFeePerGas - maxPriorityFeePerGas) / 2) as defined by ethers
+// https://github.com/ethers-io/ethers.js/blob/v5.7.2/packages/abstract-provider/src.ts/index.ts#L253
+export const getEip1559GasPrice = ({
+  fast,
+  average,
+  slow,
+}: GasFeeDataEstimate): Eip1559GasPrice => ({
+  fast:
+    fast.maxFeePerGas && fast.maxPriorityFeePerGas
+      ? bnOrZero(fast.maxFeePerGas)
+          .minus(bnOrZero(fast.maxPriorityFeePerGas))
+          .div(2)
+          .plus(fast.maxPriorityFeePerGas)
+          .toFixed(0)
+      : undefined,
+  average:
+    average.maxFeePerGas && average.maxPriorityFeePerGas
+      ? bnOrZero(average.maxFeePerGas)
+          .minus(bnOrZero(average.maxPriorityFeePerGas))
+          .div(2)
+          .plus(average.maxPriorityFeePerGas)
+          .toFixed(0)
+      : undefined,
+  slow:
+    slow.maxFeePerGas && slow.maxPriorityFeePerGas
+      ? bnOrZero(slow.maxFeePerGas)
+          .minus(bnOrZero(slow.maxPriorityFeePerGas))
+          .div(2)
+          .plus(slow.maxPriorityFeePerGas)
+          .toFixed(0)
+      : undefined,
+})
+
+export const getTxFee = (
```

Comment:
Let's LSP destructure args for clarity / to avoid passing the args in the wrong order? Arity of 3 with 3 strings is quite bug-prone, since TS won't error if we pass things in the wrong order